lib/plugins/aws/package/compile/events/s3/index.js

```diff
@@ -126,14 +165,59 @@ class AwsCompileS3Events {
     // iterate over all buckets to be created
     // and compile them to CF resources
     _.forEach(bucketsLambdaConfigurations, (bucketLambdaConfiguration, bucketName) => {
+      let bucketProps = null;
+      if (bucketsLogicalIds[bucketName]) {
+        const providerBucket = provderS3[bucketsLogicalIds[bucketName].bucket];
+        if (providerBucket) {
+          bucketProps = Object.keys(providerBucket).reduce((acc, key) => {
```

Comment:
`reduce` is a controversial and confusuing choice for such operation. Can we just iterate over entries as:

```javascript
bucketProps = {};
for (const [key, name] of _.entries(bucketConf)) {
  ...
}
```